sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -407,260 +408,284 @@ class GpuSequenceMeta(
   }
 
   override def convertToGpu(): GpuExpression = {
-    if (expr.stepOpt.isDefined) {
-      val Seq(start, stop, step) = childExprs.map(_.convertToGpu())
-      GpuSequenceWithStep(start, stop, step, expr.timeZoneId)
-    } else {
-      val Seq(start, stop) = childExprs.map(_.convertToGpu())
-      GpuSequence(start, stop, expr.timeZoneId)
-    }
+    val (startExpr, stopExpr, stepOpt) = if (expr.stepOpt.isDefined) {
+        val Seq(start, stop, step) = childExprs.map(_.convertToGpu())
+        (start, stop, Some(step))
+      } else {
+        val Seq(start, stop) = childExprs.map(_.convertToGpu())
+        (start, stop, None)
+      }
+    GpuSequence(startExpr, stopExpr, stepOpt, expr.timeZoneId)
   }
 }
 
-object GpuSequenceUtil {
+object GpuSequenceUtil extends Arm {
 
-  def numberScalar(dt: DataType, value: Int): Scalar = dt match {
-    case ByteType => Scalar.fromByte(value.toByte)
-    case ShortType => Scalar.fromShort(value.toShort)
-    case IntegerType => Scalar.fromInt(value)
-    case LongType => Scalar.fromLong(value.toLong)
-    case _ =>
-      throw new IllegalArgumentException("wrong data type: " + dt)
+  def isAllTrue(col: ColumnVector): Boolean = {
+    assert(DType.BOOL8 == col.getType)
+    if (col.getRowCount == 0) {
+      return true
+    }
+    if (col.hasNulls) {
+      return false
+    }
+    withResource(col.all()) { allTrue =>
+      // Guaranteed there is at least one row and no nulls so result must be valid
+      allTrue.getBoolean
+    }
   }
-}
-
-/** GpuSequence without step */
-case class GpuSequence(start: Expression, stop: Expression, timeZoneId: Option[String] = None)
-    extends GpuBinaryExpression with TimeZoneAwareExpression {
-
-  override def left: Expression = start
-
-  override def right: Expression = stop
-
-  override def dataType: DataType = ArrayType(start.dataType, containsNull = false)
 
-  override def withTimeZone(timeZoneId: String): TimeZoneAwareExpression =
-    copy(timeZoneId = Some(timeZoneId))
+  /** (Copied most of the part from GpuConditionalExpression) */
```

Comment:
Please refactor to reduce code duplication, as this seems to be essentially the same.  If the original `GpuConditionalExpression` code was refactored into a utility method that took `ColumnVector` arguments then I think it could be reused verbatim.

Reply:
This is no longer needed